### PR TITLE
chore: break down dbauthz.System into smaller roles

### LINKED ISSUE #6158: chore: coderd: refactor dbauthz.AsSystem to individual roles
https://github.com/coder/coder/pull/5919 adds `dbauthz.AsSystem` which provides a code path for "sudo" access to the database. Ideally we don't want an all-powerful system actor hanging around, so we want to break this up into smaller individual roles for the following:
- provisionerd
- autostart
- ...

@Emyrk

### PR DESCRIPTION
**Pull request by johnstcn**
- rbac: export rbac.Permissions
- dbauthz: move GetDeploymentDAUs, GetTemplateDAUs, GetTemplateAverageBuildTime from querier.go to system.go and removes auth checks
- dbauthz: remove AsSystem(), add invididual roles for metrics cache, autostart, provisionerd, add restricted system role for everything else

Fixes https://github.com/coder/coder/issues/6158

### CODE REVIEW

File: coderd/database/dbauthz/dbauthz.go [hunk 1]
```diff
@@ -117,29 +117,94 @@ func ActorFromContext(ctx context.Context) (rbac.Subject, bool) {
...
+			},
+		}),
+		Scope: rbac.ScopeAll,
+	},
+	)
+}
+
+// AsSystemRestricted returns a context with an actor that has permissions
+// required for various system operations e.g. login, logout.
+func AsSystemRestricted(ctx context.Context) context.Context {
```
Comment (Emyrk) [inline coderd/database/dbauthz/dbauthz.go:189]: Do we plan on keeping this around? Or is it a catch all for the remaining stuff for now?

File: enterprise/coderd/coderd_test.go [hunk 2]
```diff
@@ -201,3 +204,23 @@ func TestAuditLogging(t *testing.T) {
 		assert.Equal(t, reflect.ValueOf(ea).Type(), reflect.ValueOf(auditor).Type())
 	})
 }
+
+// testDBAuthzRole returns a context with a subject that has a role
+// with permissions required for test setup.
+func testDBAuthzRole(ctx context.Context) context.Context {
```
Comment (Emyrk) [inline enterprise/coderd/coderd_test.go:210]: This should probably just be wildcard.wildcard? Since it's for testing.

**Review by Emyrk — APPROVED** [review]
On the right path

File: coderd/database/dbauthz/dbauthz.go (hunk 1 above)
Comment (johnstcn) [reply coderd/database/dbauthz/dbauthz.go:189]: This is a catch-all for the remaining stuff. It's mostly used for HTTP middleware.
I've pared down all the perms except read, which can still be pared down to the bare minimum if need be.

If we need to break it down further in future, we can do so. I think this is fine for now though.

**Pull request merged**